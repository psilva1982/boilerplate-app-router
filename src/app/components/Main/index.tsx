import * as S from './styles'

const Main = () => (
  <S.Wrapper>
    <main>
      <h1>React avanÃ§ado</h1>
    </main>
  </S.Wrapper>
)

export default Main
