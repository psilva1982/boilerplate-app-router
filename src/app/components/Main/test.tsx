import { render, screen } from '@testing-library/react'
import Main from '.'

describe('<Main />', () => {
  it('should render the heading', () => {
    // Renderiza o componente
    const { container } = render(<Main />)
    // Busca o elemento
    expect(
      screen.getByRole('heading', { name: /react avançado/i })
    ).toBeInTheDocument()
    // Faz comparação
    expect(container.firstChild).toMatchSnapshot()
  })
})
