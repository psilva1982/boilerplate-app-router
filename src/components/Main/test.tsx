import { render, screen } from '@testing-library/react'
import Main from '.'

describe('<Main />', () => {
  it('should render the heading', () => {
    // Renderiza o componente
    const { container } = render(<Main />)
    // Busca o elemento
    expect(
      screen.getByRole('heading', { name: /react avançado/i })
    ).toBeInTheDocument()
    // Faz comparação
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the colors correctly', () => {
    // renderiza o component
    const { container } = render(<Main />)

    // verifica se o background-color está correto
    expect(container.firstChild).toHaveStyle({ 'background-color': '#06092b' })
  })
})
