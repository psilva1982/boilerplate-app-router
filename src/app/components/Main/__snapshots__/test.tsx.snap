// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
<main
  class="sc-bcPKhP jhrJfq"
>
  <main>
    <h1>
      React avan√ßado
    </h1>
  </main>
</main>
`;
